crates/bevy_core/src/serde.rs

```diff
@@ -18,7 +21,7 @@ impl Serialize for Name {
 
 impl<'de> Deserialize<'de> for Name {
     fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
-        deserializer.deserialize_str(EntityVisitor)
+        deserializer.deserialize_any(EntityVisitor)
```

Comment:
I could be wrong but I believe this is also a breaking change. IIRC non-self-describing formats do not support `deserialize_any`.

Could you verify that this doesn't break non-self-describing formats?